src/lib/annotations/drawing/DrawingThread.js

```diff
@@ -0,0 +1,131 @@
+    lastRenderTimestamp;
+
+    /** @property {number} - Drawing state */
+    drawingFlag;
+
+    /** @property {Rbush} - Rtree path container */
+    pathContainer;
+
+    /** @property {ImageData} - The ImageData after the last saved stroke */
+    savedState;
+
+    /** @property {DrawingPath} - The path being drawn but not yet finalized */
+    pendingPath;
+
+    /** @property {CanvasContext} - The context to be drawn on */
+    context;
+
+    //--------------------------------------------------------------------------
+    // Public
+    //--------------------------------------------------------------------------
+    /**
+     * Creates and mananges drawing annotations
+     *
+     * [constructor]
+     * @inheritdoc
+     * @return {DrawingThread} DrawingThread
+     */
+    constructor(data) {
+        super(data);
+
+        this.drawingFlag = DRAW_POINTER_UP;
+        this.pathContainer = new Rbush(RTREE_WIDTH);
+    }
+
+    /**
+     * Soft destroy a drawingthread object
+     *
+     * [constructor]
+     * @inheritdoc
+     * @return {void}
+     */
+    destroy() {
+        this.removeAllListeners();
+        this.reset();
+        super.destroy();
+    }
+
+    /**
+     * Get all of the DrawingPaths in the current thread.
+     * @return {void}
+     */
+    getDrawings() {
+        return this.pathContainer.data.children;
+    }
+
+    /* eslint-disable no-unused-vars */
+    /**
+     * Handle a pointer movement
+     *
+     * @param {Object} location The location information of the pointer
+     * @return {void}
+     */
+    handleMove(location) {}
+
+    /**
+     * Start a drawing stroke
+     *
+     * @param {Object} location The location information of the pointer
+     * @return {void}
+     */
+    handleStart(location) {}
+
+    /**
+     * End a drawing stroke
+     *
+     * @param {Object} location The location information of the pointer
+     * @return {void}
+     */
+    handleStop(location) {}
+    /* eslint-disable no-unused-vars */
+
+    //--------------------------------------------------------------------------
+    // Protected
+    //--------------------------------------------------------------------------
+
+    /**
+     * @protected
+     * @param {number} scale The document scale
+     * @return {void}
+     */
+    updateContextStyles(scale) {
+        if (this.context) {
+            this.context.lineCap = 'round';
+            this.context.lineJoin = 'round';
+            this.context.strokeStyle = 'black';
+            this.context.lineWidth = BASE_LINE_WIDTH * scale;
+        }
+    }
+
+    /**
+     * @protected
+     * @param {number} timestamp The time when the function was called;
+     * @return {void}
+     */
+    render(timestamp) {
+        if (this.drawingFlag === DRAW_POINTER_DOWN) {
+            const elapsed = timestamp - (this.lastRenderTimestamp || 0);
+            if (elapsed >= DRAW_RENDER_THRESHOLD && this.context) {
```

Comment:
Also, pair this delta time check with input events, to have an even MORE performant render loop. There's no sense in re-rendering the whole scene if you're just sitting with the pointer in a "down" state